docs/concepts/linq.rst

```diff
@@ -0,0 +1,268 @@
+	
+	// Using the query syntax
+	cats = from dog in dogs
+	       select dog.TurnIntoACat();
+	
+	// Reducing a list of type A to a single result of type A
+	var frankensteinDog = dogs.Aggregate((current, next) => MadScientist.StitchTogether(current, next));
+	
+* Flattening a list of lists:
+
+.. code-block:: c#
+
+	// Transforms the list of kennels into a list of all their dogs.
+	var allDogsFromKennels = kennels.SelectMany(kennel => kennel.Dogs);
+	
+* Union between two sets (with custom comparator)!
+
+.. code-block:: c#
+
+	public class DogHairLengthComparer : IEqualityComparer<Dog>
+	{
+	    public bool Equals(Dog a, Dog b)
+	    {
+	        if (a == null && a == null)
+	        {
+	            return true;
+	        }
+	        else if ((a == null && b != null) ||
+	    	         (a != null && b == null))
+	        {
+	            return false;
+	        }
+	        else
+	        {
+	            return a.HairLengthType == b.HairLengthType;
+	        }
+	    }
+		
+	    public int GetHashCode(Dog d)
+	    {
+	        // default hashcode is enough here, as these are simple objects.
+	        return b.GetHashCode();
+	    }
+	}
+	
+	...
+	
+	// Gets all the short-haired dogs between two different kennels
+	var allShortHairedDogs = kennel1.Dogs.Union(kennel2.Dogs, new DogHairLengthComparer());
+	
+* Intersection between two sets!
+
+.. code-block:: c#
+
+	// Gets the volunteers who spend share time with two humane societies.
+	var volunteers = humaneSociety1.Volunteers.Intersect(humaneSociety2.Volunteers,
+	                                                     new VolunteerTimeComparer());
+														 
+* Ordering!
+
+.. code-block:: c#
+
+	// Get driving directions, ordering by if it's toll-free before estimated driving time.
+	var results = DirectionsProcessor.GetDirections(start, end)
+	              .OrderBy(direction => direction.HasNoTolls)
+	              .ThenBy(direction => direction.EstimatedTime);
+				  
+Finally, a more advanced sample: determining if two objects are "shallow" equal:
+
+.. code-block:: c#
+
+	var dog1Props = dog1.GetType().GetProperties();
+	var dog2Props = dog2.GetType().GetProperties();
+
+	// Note that if the Dog type contains any nested objects,
+	// or what's commonly referred to an "object graph",
+	// this query will fail.
+	var unequalProps = from prop1 in dog1Props
+	                   let  prop1Value = prop1.GetValue(object1)
+	                   let  prop2 = dog2Props.GetProperty(prop1.Name)
+	                   let  prop3value = prop2.GetValue(object2)
+	                   where !prop1Value.Equals(prop2Value)
+	                   select new { Prop =  prop1.Name, prop1Value, prop2Value }
+	
+	if (unequalProps.Any())
+	{
+	    // The two objects differ!
+	    foreach (var unequalSet in unequalProps)
+	    {
+	        Console.WriteLine($"Unequal Property {unequalSet.Prop} - {prop1Value} != {prop2value}");
+	    }
+	}
+				  
+Why Use the Query Syntax?
```

Comment:
Though I like this section, I feel it needs to be better written. I never found query syntax to be "more powerful", but then again, I never had the need/want to compare the two. If "let" is the only thing that is different, wouldn't it be better to say "Differences in two ways of writing LINQ queries"?